.clang-tidy

```diff
@@ -0,0 +1,9 @@
+---
+Checks: >
+    *-,
+    cppcoreguidelines-macro-usage,
+    modernize-use-nullptr,
+    modernize-use-override,
+    readability-avoid-const-params-in-decls
```

Comment:
If I understand it correctly, this will disallow adding const when parameters are passed by value anyway,  like `const uint8_t spam`. Is that correct?

Reply:
Also that reminded me to have https://clang.llvm.org/extra/clang-tidy/checks/readability/const-return-type.html added in the next batch of checks.

Reply:
Can we have the opposite. It's better to have the const in the declaration then you know if a parameter is modified. The definition can drop the const if they want since it isn't part of the signature.

Reply:
The parameter can not be modified unless its a reference, for references it won't say anything.

Reply:
@ZehMatt That is not quite what I asked... does it _only_ affect cases where the parameter is passed by value? I already found the page you linked to myself, but didn’t find it clear.

Reply:
Here is a more concrete example to when it applies:
```cpp
// Complains because const is meaningless here
void func(const int arg1);

// Won't complain.
void func(const int& arg1);

// Won't complain, here const makes sense if you don't want it to mutate within the function body, still matches the signature of void func(const int arg1);
void func(const int arg1) { ... }
```

Reply:
Yes so I'd like it to be the opposite. It should complain on `void func(int arg1);` to improve readability you want const to be on all interfaces. Then a non const reference is very obvious. The implementation can drop the const if you want but don't care as much.

Reply:
But having const in the declaration does nothing really, when its passed by value then there are no side effects, so why do you want const there?

Reply:
> Here is a more concrete example to when it applies:
> 
> ```c++
> // Complains because const is meaningless here
> void func(const int arg1);
> 
> // Won't complain.
> void func(const int& arg1);
> 
> // Won't complain, here const makes sense if you don't want it to mutate within the function body, still matches the signature of void func(int arg1);
> void func(const int arg1) { ... }
> ```

So does it leave classes and structs as parameters alone?

Reply:
When its a const reference yes.

Reply:
I'd rather we don't enable this particular rule.

Reply:
I'll remove it.